### ISSUE TITLE
Type declaration not correctly implemented

### ISSUE THREAD

**Issue by jlmakes** [jlmakes/tealight#1]
```
Could not find a declaration file for module 'tealight'.
'.../node_modules/tealight/dist/tealight.js' implicitly has an 'any' type.
Try `npm install @types/tealight` if it exists or add a new declaration (.d.ts) file containing `declare module 'tealight';`
```

Seems the issue was `index.d.ts` not being included in the published files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)

**Comment by jlmakes** [comment]
Seems like the issue was two fold:

- The `index.d.ts` file was not included in the published package files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)
- When importing the module, it should use the CommonJS syntax, e.g:

```ts
import tealight = require("tealight");
```

Maybe there's a way to use ES2015 syntax, but I'm still new to TypeScript, and it appears to be working so I'll consider this solved for now.

**Comment by jlmakes** [comment]
Seems the issue was that the `index.d.ts` file was not included in the published files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)

**Comment by jlmakes** [comment]
Seems the issue was `index.d.ts` was not included in the published files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)

**Comment by jlmakes** [comment]
Seems the issue was `index.d.ts` was not included in the published files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)

Also, I'm not so sure about the different combations of `compilerOptions.module` and how it determines what are valid `import` statements.

- Omitting the `tsconfig.json` altogether (using Parcel) allows `import tealight from "tealight";`
- `compilerOptions.module = "es2015"` won’t allow `import tealight from "tealight";`
- `compilerOptions.module = "commonjs"` allows `import tealight = require("tealight");`

TypeScript itself is fairly intuitive, but the workflow is incredibly opaque 😅

**Comment by jlmakes** [comment]
Seems the issue was `index.d.ts` was not included in the published files (fixed in cf14e523351963aa9b7939397f5aac1d3dd3829b)
